benchmark/source_map/source-map-cache.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const common = require('../common.js');
+const assert = require('assert');
+const path = require('path');
+
+const options = {
+  flags: ['--expose-internals'],
+};
+
+const bench = common.createBenchmark(
+  main,
+  {
+    operation: [
+      'findSourceMap-valid',
+      'findSourceMap-generated-source',
+    ],
+    n: [1e5],
+  },
+  options,
+);
+
+function main({ operation, n }) {
+  const Module = require('node:module');
+
+  Module.setSourceMapsSupport(true);
+  const validFileName = path.resolve(
+    __dirname,
+    '../../test/fixtures/test-runner/source-maps/line-lengths/index.js',
+  );
+
+  const sourceWithGeneratedSourceMap = `
```

Comment:
FWIW, this can be read from the file and appending the `//# sourceURL` magic comment to read source, so that we don't have to inline it here.

Reply:
Good one, thanks!